File: types/CodeEditorTypes.go
package types

type EditorChange struct {
	Type   string     `json:"type"`
	Path   string     `json:"path"`
	Change ChangeData `json:"change"`
}

type ChangeData struct {
	Type      string      `json:"type"`
	Range     ChangeRange `json:"range"`
	Text      string      `json:"text"`
	Timestamp int64       `json:"timestamp"`
}

type ChangeRange struct {
	StartLineNumber int `json:"startLineNumber"`
	StartColumn     int `json:"startColumn"`
	EndLineNumber   int `json:"endLineNumber"`
	EndColumn       int `json:"endColumn"`
}

File: web/src/types/CodeEditorTypes.ts
interface WebSocketResponseType {
    type: string,
    error?: string,
    res?: string,
    path?: string,
    totalSize?: string,
    isCompleted?: boolean,
    abortMsg?: string,
    change?: ChangeData
}

interface EditorChange {
    type: string,
    path: string,
    count?: number,
    content?: string,
    error?: string,
    change: ChangeData
}

interface ChangeData {
    type: string,
    range: ChangeRange,
    text: string,
    timestamp: number,
    content?: string
}

interface ChangeRange {
    startLineNumber: number,
    startColumn: number,
    endLineNumber: number,
    endColumn: number
}

export {type WebSocketResponseType, type EditorChange, type ChangeData, type ChangeRange}